AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthenticationProviderAdapter+SignIn.swift

```diff
@@ -74,6 +68,45 @@ extension AuthenticationProviderAdapter {
         }
     }
 
+    func confirmSignIn(request: AuthConfirmSignInRequest,
+                       completionHandler: @escaping (Result<AuthSignInResult, AmplifyAuthError>) -> Void) {
+
+        let userAttributes = (request.options.pluginOptions as? AWSAuthConfirmSignInOptions)?.userAttributes
+        let clientMetaData = (request.options.pluginOptions as? AWSAuthConfirmSignInOptions)?.metadata
+        awsMobileClient.confirmSignIn(challengeResponse: request.challengeResponse,
+                                      userAttributes: [:],
+                                      clientMetaData: clientMetaData ?? [:]) { [weak self] result, error in
+                                        guard let self = self else { return }
+
+                                        guard error == nil else {
+                                            let result = self.convertSignInErrorToResult(error!)
```

Comment:
[low] i _think_ we wanted to avoid force cast in all cases, and to do this nicely here would be to just use `if let error = error {`

Reply:
Changed